src/applications/caregivers/components/SubmitError/DownloadLink.jsx

```diff
@@ -1,64 +1,88 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
+import * as Sentry from '@sentry/browser';
 import moment from 'moment';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
 
 import formConfig from 'applications/caregivers/config/form';
 import environment from 'platform/utilities/environment';
+import recordEvent from 'platform/monitoring/record-event';
 import { submitTransform } from 'applications/caregivers/helpers';
 import { apiRequest } from 'platform/utilities/api';
+import { isValidForm } from 'platform/forms-system/src/js/validation';
+import { createFormPageList } from 'platform/forms-system/src/js/helpers';
 
 const DownLoadLink = ({ form }) => {
   const [PDFLink, setPDFLink] = useState(null);
+  const [isLoading, setLoading] = useState(false);
   const formData = submitTransform(formConfig, form);
   const veteranFullName = form.data.veteranFullName;
+  const pageList = createFormPageList(formConfig);
+  const isFormValid = isValidForm(form, pageList);
 
-  useEffect(
-    () => {
-      apiRequest(
-        `${environment.API_URL}/v0/caregivers_assistance_claims/download_pdf`,
-        {
-          method: 'POST',
-          body: formData,
-          headers: {
-            'Content-Type': 'application/json',
-            'Source-App-Name': 'caregivers-10-10cg-',
-          },
+  const downloadPDF = () => {
+    setLoading(true);
+    apiRequest(
+      `${environment.API_URL}/v0/caregivers_assistance_claims/download_pdf`,
+      {
+        method: 'POST',
+        body: formData,
+        headers: {
+          'Content-Type': 'application/json',
+          'Source-App-Name': 'caregivers-10-10cg-',
         },
-      )
-        .then(response => {
-          return response.blob();
-        })
-        .then(blob => {
-          const url = URL.createObjectURL(blob);
-          setPDFLink(url);
-        });
-    },
-    [formData],
-  );
+      },
+    )
+      .then(response => response.blob())
+      .then(blob => {
+        const url = URL.createObjectURL(blob);
+        setPDFLink(url);
+        setLoading(false);
+        recordEvent({ event: 'caregivers-10-10cg-pdf-download--success' });
+      })
+      .catch(error => {
+        setLoading(false);
```

Comment:
It looks like we don't provide the user with any feedback when the API request fails. We would consider providing a message that "We could not generate your PDF. Please try again."

It would be confusing the click and see nothing happen.

Reply:
This is a question for @jonathanjnelson

Reply:
To be clear I dont think this is really possible since the UI validates all the data... and does not let you save unless its valid...

Reply:
I was more referencing a 5xx error, not a bad request from the client.

If the data is valid, the request is made, but API broke or is down, will the user be informed something went wrong?

Maybe there is something globally that would notify the user, but from this code, it looks like they'd click the button, it would load, then stop loading, and nothing will happen.

Reply:
@kevinmirc make it disable with cursor of not-allowed also clear the error once error is cleared. Although I think the error will be very unlikely to hit with the UI validation being double checked onFetch.

Reply:
The error message **Error downloading 1010-cg PDF** has a few of issues (in my opinion):

- It's being used when the data is invalid (client error) but it reads like a server error (an error that is out of the control of the user to fix). I feel that a message informing the users it's their fault, not a server error would be helpful.
- See a typo: "1010-cg" should be "10-10CG"
- This still doesn't notify the user of a 5xx error coming back from the API

I'm approving since this resolved the original bug and the UX/UI can be a follow-up enhancement.